fc/protocol.py

```diff
@@ -81,6 +81,19 @@ def __init__(self, proto_file, indent_level=0):
 
         The protocol must be specified using the textual syntax, as defined by the CompactSyntaxParser module.
         """
+        # if the filename passed as argument to CompactSyntaxParser is not found
+        # it tries to read it before checking it exists
+        # and you get a messy exit
+        # file_contents = file_or_filename.read()
+        #     E AttributeError: 'str' object has no attribute 'read'
+        #
+        #     During handling of the above exception, another exception occurred:
+        #     E FileNotFoundError: [Errno 2] No such file or directory:
+        # throw exception
+
+        with open(proto_file) as f:
```

Comment:
Would it be better to do something like `assert os.path.exists(proto_file)`? That doesn't check you can read it, but seems cleaner than just trying to open the file! Or put more checks where the file is read?

Reply:
I was specifically going for getting the FileNotFound exception rather a generic one but will defer to whichever you wish :-)

Reply:
Fair enough. And actually I'm not sure there's a simpler way to check for readability of a file!

Reply:
Putting the checks where the file is read is difficult as the CompactSyntaxParser(CSP) code uses pyparsing to do this and in a fairly inconsistent way where is tries to open before checking it exists and gets an exception that throws an exception. 
I was trying to avoid needing to look closer at CSP.
Also when looking for a library file I need to do this catch before it tries to do anything with the protocol. Locating the check at the beginning of Protocol init catches both cases neatly; trying to hack CSP does not.